src/descriptor/satisfied_contraints.rs

```diff
@@ -0,0 +1,1610 @@
+                }
+                AstElem::Thresh(k, ref subs) if node_state.n_evaluated == subs.len() => {
+                    if let Some(res) = self.stack.pop() {
+                        match res {
+                            StackElement::Dissatisfied if node_state.n_satisfied == k =>
+                                self.stack.push(StackElement::Satisfied),
+                            StackElement::Satisfied if node_state.n_satisfied == k - 1 =>
+                                self.stack.push(StackElement::Satisfied),
+                            StackElement::Satisfied | StackElement::Dissatisfied =>
+                                self.stack.push(StackElement::Dissatisfied),
+                            StackElement::Push(_v) =>
+                                return Some(Err(Error::UnexpectedStackElementPush))
+                        }
+                    } else {
+                        return Some(Err(Error::UnexpectedStackEnd))
+                    }
+                }
+                AstElem::Thresh(ref _k, ref subs) if node_state.n_evaluated != 0 => {
+                    if let Some(res) = self.stack.pop() {
+                        match res {
+                            StackElement::Dissatisfied => {
+                                self.state.push(NodeEvaluationState {
+                                    node: node_state.node,
+                                    n_evaluated: node_state.n_evaluated + 1,
+                                    n_satisfied: node_state.n_satisfied
+                                });
+                            }
+                            StackElement::Satisfied => {
+                                self.state.push(NodeEvaluationState {
+                                    node: node_state.node,
+                                    n_evaluated: node_state.n_evaluated + 1,
+                                    n_satisfied: node_state.n_satisfied + 1
+                                });
+                            }
+                            StackElement::Push(_v) =>
+                                return Some(Err(Error::UnexpectedStackElementPush))
+                        }
+                        self.state.push(NodeEvaluationState {
+                            node: &subs[node_state.n_evaluated],
+                            n_evaluated: 0,
+                            n_satisfied: 0
+                        });
+                    } else {
+                        return Some(Err(Error::UnexpectedStackEnd))
+                    }
+                }
+                AstElem::ThreshM(ref k, ref subs) if node_state.n_evaluated == 0 => {
+                    let len = self.stack.len();
+                    if len < k + 1 {
+                        return Some(Err(Error::InsufficientSignaturesMultiSig))
+                    } else {
+                        //Non-sat case. If the first sig is empty, others k elements must
+                        //be empty.
```

Comment:
I don't see why this should be true.

Reply:
MultiSig takes k-sigs all of which must be valid or all must be empty. Anything else aborts the script as per standardness.

Reply:
Oh, ok, fair enough

Reply:
For reference: 
https://github.com/bitcoin/bitcoin/blob/0822b44d8a68a3f29b12719523f6cb5253eb679c/src/script/interpreter.cpp#L1034